From b8795a16a131d5576d2378eb7db7eb240119a2cf Mon Sep 17 00:00:00 2001
From: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
Date: Mon, 7 Apr 2014 19:46:50 +0530
Subject: [PATCH 341/793] spi: zynq-qspi: Clean up isr

Cleanup ISR for readability.
Dont disable interrupts and re-enable again.

Signed-off-by: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
Signed-off-by: Harini Katakam <harinik@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/spi/spi-zynq-qspi.c |   12 +-----------
 1 files changed, 1 insertions(+), 11 deletions(-)

diff --git a/drivers/spi/spi-zynq-qspi.c b/drivers/spi/spi-zynq-qspi.c
index 63de2d6..72e5eb4 100644
--- a/drivers/spi/spi-zynq-qspi.c
+++ b/drivers/spi/spi-zynq-qspi.c
@@ -439,8 +439,6 @@ static irqreturn_t zynq_qspi_irq(int irq, void *dev_id)
 
 	intr_status = zynq_qspi_read(xqspi->regs + ZYNQ_QSPI_STATUS_OFFSET);
 	zynq_qspi_write(xqspi->regs + ZYNQ_QSPI_STATUS_OFFSET , intr_status);
-	zynq_qspi_write(xqspi->regs + ZYNQ_QSPI_IDIS_OFFSET,
-			ZYNQ_QSPI_IXR_ALL_MASK);
 
 	if ((intr_status & ZYNQ_QSPI_IXR_TXNFULL_MASK) ||
 		   (intr_status & ZYNQ_QSPI_IXR_RXNEMTY_MASK)) {
@@ -496,18 +494,10 @@ static irqreturn_t zynq_qspi_irq(int irq, void *dev_id)
 					zynq_qspi_write(xqspi->regs +
 						offset[tmp - 1], data);
 			}
-			zynq_qspi_write(xqspi->regs + ZYNQ_QSPI_IEN_OFFSET,
-					ZYNQ_QSPI_IXR_ALL_MASK);
 		} else {
 			/* If transfer and receive is completed then only send
 			 * complete signal */
-			if (xqspi->bytes_to_receive) {
-				/* There is still some data to be received.
-				   Enable Rx not empty interrupt */
-				zynq_qspi_write(xqspi->regs +
-						ZYNQ_QSPI_IEN_OFFSET,
-						ZYNQ_QSPI_IXR_ALL_MASK);
-			} else {
+			if (!xqspi->bytes_to_receive) {
 				zynq_qspi_write(xqspi->regs +
 						ZYNQ_QSPI_IDIS_OFFSET,
 						ZYNQ_QSPI_IXR_ALL_MASK);
-- 
1.7.5.4

